### PR TITLE
[SPARK-36208][SQL] SparkScriptTransformation should support ANSI interval types

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR changes `BaseScriptTransformationExec` for `SparkScriptTransformationExec` to support ANSI interval types.

### Why are the changes needed?

`SparkScriptTransformationExec` support `CalendarIntervalType` so it's better to support ANSI interval types as well.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala [hunk 1]
```diff
@@ -223,6 +224,14 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
       case CalendarIntervalType => wrapperConvertException(
         data => IntervalUtils.stringToInterval(UTF8String.fromString(data)),
         converter)
+      case YearMonthIntervalType(start, end) => wrapperConvertException(
+        data => Period.ofMonths(
+          IntervalUtils.castStringToYMInterval(UTF8String.fromString(data), start, end)),
+        converter)
+      case DayTimeIntervalType(start, end) => wrapperConvertException(
+        data => Duration.ofNanos(IntervalUtils.castStringToDTInterval(
+          UTF8String.fromString(data), start, end) * DateTimeConstants.NANOS_PER_MICROS),
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala:233]:
```suggestion
        data => IntervalUtils.microsToDuration(
          IntervalUtils.castStringToDTInterval(UTF8String.fromString(data), start, end)),
```

File: sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala [hunk 2]
```diff
@@ -223,6 +224,14 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
       case CalendarIntervalType => wrapperConvertException(
         data => IntervalUtils.stringToInterval(UTF8String.fromString(data)),
         converter)
+      case YearMonthIntervalType(start, end) => wrapperConvertException(
+        data => Period.ofMonths(
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala:228]:
```suggestion
        data => IntervalUtils.monthsToPeriod(
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]
Please, use utils functions.

**Comment by sarutak** [comment]
@MaxGekk Thanks, got it.

**Review by MaxGekk — APPROVED** [review]
Waiting for jenkins.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.2.
Thank you, @sarutak .

**Comment by MaxGekk** [comment]
@sarutak Sorry, I haven't merged to branch-3.2. Here is the PR for it: https://github.com/apache/spark/pull/33463 @HyukjinKwon @gengliangwang Could you approve it, please.

**Pull request closed without merging**